pytensor/xtensor/rewriting/indexing.py

```diff
@@ -0,0 +1,150 @@
+        x_dims = x.type.dims
+        x_shape = tuple(x.shape)
+        out_ndim = out.type.ndim
+        out_dims = out.type.dims
+        aligned_idxs = []
+        basic_idx_axis = []
+        # zip_longest adds the implicit slice(None)
+        for i, (idx, x_dim) in enumerate(
+            zip_longest(idxs, x_dims, fillvalue=as_symbolic(slice(None)))
+        ):
+            if isinstance(idx.type, SliceType):
+                if not any(
+                    (
+                        isinstance(other_idx.type, XTensorType)
+                        and x_dim in other_idx.dims
+                    )
+                    for j, other_idx in enumerate(idxs)
+                    if j != i
+                ):
+                    # We can use basic indexing directly if no other index acts on this dimension
+                    # This is an optimization that avoids creating an unnecessary arange tensor
+                    # and facilitates the use of the specialized AdvancedSubtensor1 when possible
+                    aligned_idxs.append(idx)
+                    basic_idx_axis.append(out_dims.index(x_dim))
+                else:
+                    # Otherwise we need to convert the basic index into an equivalent advanced indexing
+                    # And align it so it interacts correctly with the other advanced indices
+                    adv_idx_equivalent = arange(x_shape[i])[to_basic_idx(idx)]
+                    ds_order = ["x"] * out_ndim
+                    ds_order[out_dims.index(x_dim)] = 0
+                    aligned_idxs.append(adv_idx_equivalent.dimshuffle(ds_order))
+            else:
+                assert isinstance(idx.type, XTensorType)
+                if idx.type.ndim == 0:
+                    # Scalar index, we can use it directly
+                    aligned_idxs.append(idx.values)
+                else:
+                    # Vector index, we need to align the indexing dimensions with the base_dims
+                    ds_order = ["x"] * out_ndim
+                    for j, idx_dim in enumerate(idx.dims):
+                        ds_order[out_dims.index(idx_dim)] = j
+                    aligned_idxs.append(idx.values.dimshuffle(ds_order))
+
+        # Squeeze indexing dimensions that were not used because we kept basic indexing slices
+        if basic_idx_axis:
+            aligned_idxs = [
+                idx.squeeze(axis=basic_idx_axis)
+                if (isinstance(idx.type, TensorType) and idx.type.ndim > 0)
+                else idx
+                for idx in aligned_idxs
+            ]
+
+        x_tensor_indexed = x_tensor[tuple(aligned_idxs)]
+
+        if basic_idx_axis and _non_consecutive_adv_indexing(aligned_idxs):
+            # Numpy moves advanced indexing dimensions to the front when they are not consecutive
+            # We need to transpose them back to the expected output order
+            x_tensor_indexed_basic_dims = [out_dims[idx] for idx in basic_idx_axis]
```

Comment:
Very minor nit, but I would use something other than `idx` here, maybe `axis`? My first reaction was surprise because I still had `idx` "stored in my mind map" as the `idx` variable defined in the for loop above. It is a local variable so not necessarily rational on my part but figured I'd mention.

Reply:
Yeah that's a good suggestion

Reply:
Done